pkg/runtime/runtime.go

```diff
@@ -2897,14 +2936,77 @@ func (a *DaprRuntime) getAppHTTPChannelConfig(pipeline httpMiddleware.Pipeline,
 		MaxRequestBodySizeMB: a.runtimeConfig.maxRequestBodySize,
 	}
 
-	if !isExternal {
-		conf.Endpoint = a.getAppHTTPEndpoint()
-		conf.Client = a.appHTTPClient
-	} else {
-		conf.Client = nethttp.DefaultClient
+	conf.Endpoint = a.getAppHTTPEndpoint()
+	conf.Client = a.appHTTPClient
+	return conf
+}
+
+func (a *DaprRuntime) getHTTPEndpointAppChannel(pipeline httpMiddleware.Pipeline, endpoint httpEndpointV1alpha1.HTTPEndpoint) (httpChannel.ChannelConfiguration, error) {
+	conf := httpChannel.ChannelConfiguration{
+		CompStore:            a.compStore,
+		MaxConcurrency:       a.runtimeConfig.appConnectionConfig.MaxConcurrency,
+		Pipeline:             pipeline,
+		TracingSpec:          a.globalConfig.Spec.TracingSpec,
+		MaxRequestBodySizeMB: a.runtimeConfig.maxRequestBodySize,
 	}
 
-	return conf
+	var tlsConfig *tls.Config
+
+	if endpoint.Spec.TLSRootCA.Value.String() != "" {
```

Comment:
question: what happens when the underlying client certificate or other TLS field values change- will the TLS config be updated?